### PR TITLE
Tests for ripe-pickers

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | Need for ripe-pickers tests |
| Dependencies | -- |
| Decisions | • Improved `should instantiate the component` test <br>• Added `should sync selects options` test <br><br>**NOTE:** The last webpack update broke the tests, so please `npm i webpack@4.43.0` |
| Screenshot | imagem |

### CODE REVIEW

File: vue/test/components/ripe-pickers/ripe-pickers.test.js [hunk 1]
```diff
@@ -1,19 +1,158 @@
...
 
-describe("RipePickers", () => {
-    it("should instantiate the component", () => {
-        const ripeInstance = new ripe.Ripe("dummy", "cube", {
-            version: 52
-        });
-        const component = base.getComponent("RipePickers", {
-            props: {
-                ripe: ripeInstance
+class MockRipeSdk {
```
Comment (gcandal) [inline vue/test/components/ripe-pickers/ripe-pickers.test.js]: Try to mimic the structure of https://github.com/ripe-tech/ripe-white/blob/master/vue/test/components/base-personalization-plugin/base-personalization-plugin.test.js#L8 so that we have a "base" mock with some commonly needed stuff.

File: vue/test/components/ripe-pickers/ripe-pickers.test.js [hunk 2]
```diff
@@ -1,19 +1,158 @@
...
+}
+
+describe("RipePickers", () => {
+    it("should instantiate the component", async () => {
+        const ripeInstance = new MockRipeSdk();
+        const component = await base.getComponent("RipePickers", { props: { ripe: ripeInstance } });
 
         assert.strictEqual(component.emitted("loading").length, 1);
+
+        await component.vm.$forceUpdate();
```
Comment (gcandal) [inline vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: Why is this needed?

File: vue/test/components/ripe-pickers/ripe-pickers.test.js [hunk 3]
```diff
@@ -1,19 +1,158 @@
...
+    });
+
+    it("should sync selects options", async () => {
+        const ripeInstance = new MockRipeSdk();
+        const component = await base.getComponent("RipePickers", { props: { ripe: ripeInstance } });
+
+        await component.vm.$forceUpdate();
+        assert.strictEqual(component.find(".select.parts").findAll("option").length, 4);
+        assert.strictEqual(component.find(".select.materials").findAll("option").length, 1);
+        assert.strictEqual(component.find(".select.colors").findAll("option").length, 1);
```
Comment (gcandal) [inline vue/test/components/ripe-pickers/ripe-pickers.test.js]: Move to `should instantiate the component`

File: vue/test/components/ripe-pickers/ripe-pickers.test.js [hunk 4]
```diff
@@ -1,19 +1,158 @@
...
+
+        await component.setData({ selectedPart: "side" });
```
Comment (gcandal) [inline vue/test/components/ripe-pickers/ripe-pickers.test.js]: Use the component's public API instead. It should be `selectPart("side")` but we will have to settle for `onSelectPartChange` since it's what exists (this is why we should always separate handlers from actual methods).

File: vue/test/components/ripe-pickers/ripe-pickers.test.js [hunk 5]
```diff
@@ -1,19 +1,158 @@
...
+        assert.strictEqual(component.find(".select.parts").exists(), true);
+        assert.strictEqual(component.find(".select.materials").exists(), true);
+        assert.strictEqual(component.find(".select.colors").exists(), true);
+    });
+
+    it("should sync selects options", async () => {
+        const ripeInstance = new MockRipeSdk();
+        const component = await base.getComponent("RipePickers", { props: { ripe: ripeInstance } });
+
+        await component.vm.$forceUpdate();
```
Comment (gcandal) [inline vue/test/components/ripe-pickers/ripe-pickers.test.js:115]: Why is this needed?

File: vue/components/organisms/ripe-pickers/ripe-pickers.vue [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <template>
     <div class="ripe-pickers">
         <select
-            class="select"
+            class="select parts"
```
Comment (gcandal) [inline vue/components/organisms/ripe-pickers/ripe-pickers.vue]: select-parts

**Review by gcandal — CHANGES_REQUESTED** [review]
Nice 👍

File: vue/test/components/ripe-pickers/ripe-pickers.test.js (hunk 2 above)
Comment (NFSS10) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: Because the `pre_config` and `post_config` events aren't binded in the test, so no component update. We need to manually do it ourselves.

Comment (NFSS10) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: `$nextTick` would also do the trick

File: vue/test/components/ripe-pickers/ripe-pickers.test.js (hunk 5 above)
Comment (NFSS10) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:115]: https://github.com/ripe-tech/ripe-sdk-components-vue/pull/36#discussion_r518688029

File: vue/test/components/ripe-pickers/ripe-pickers.test.js (hunk 2 above)
Comment (gcandal) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: What do you mean with "aren't binded"? The solution may end up being nextTick, but I really need more detail before settling with that.

Comment (NFSS10) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: The parts, materials and colors values for the selects are computed, so we need to trigger something that can lead to a component rerender.

Comment (gcandal) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: Isn't there a more high-level way of doing this? E.g.: triggering an event, mutating the store...

Comment (NFSS10) [reply vue/test/components/ripe-pickers/ripe-pickers.test.js:102]: I don't think so because it's a vue-test issue

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]
